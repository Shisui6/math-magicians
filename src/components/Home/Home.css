.home {
  padding: 85px 100px;
  height: calc(100vh - 72px);
  background-image: url('../../../public/home1.jpg');
  background-size: 100% 100%;
  background-position: center;
  background-repeat: no-repeat;
  opacity: 0;
  animation: fade-in 0.8s forwards;
}

@keyframes fade-in {
  100% {
    opacity: 1;
  }
}

.home h1 {
  font-size: 50px;
}

.home p {
  width: 90%;
  line-height: 28px;
}

@media screen and (max-width: 768px) {
  .home p {
    width: 100%;
    line-height: 28px;
  }

  .home {
    padding: 100px 50px;
  }
}
