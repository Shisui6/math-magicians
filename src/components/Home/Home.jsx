import './Home.css';

const Home = () => (
  <div className="home">
    <h1>Hello Traveler!</h1>
    <p>
      Welcome to Math Magicians! This app is designed to make
      your life easier by allowing you to quickly and easily calculate anything you need.
      Whether you&apos;re trying to figure out a math problem or solve a complex equation,
      this app will do the trick. So go ahead, click calculate and solve those problems!
    </p>
  </div>
);

export default Home;
