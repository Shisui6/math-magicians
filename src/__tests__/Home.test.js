import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Home from '../components/Home/Home';

describe('Home', () => {
  it('renders correctly', () => {
    const component = renderer.create(<Home />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('contains the p element', () => {
    render(<Home />);

    expect(screen.getByText('Hello Traveler!')).toBeInTheDocument();
  });
});
