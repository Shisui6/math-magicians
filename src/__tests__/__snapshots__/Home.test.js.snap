// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<div
  className="home"
>
  <h1>
    Hello Traveler!
  </h1>
  <p>
    Welcome to Math Magicians! This app is designed to make your life easier by allowing you to quickly and easily calculate anything you need. Whether you're trying to figure out a math problem or solve a complex equation, this app will do the trick. So go ahead, click calculator and solve those problems!
  </p>
</div>
`;
